File: src/metrics/emergence/initialise.java
public void initialise(){
		initialise(base, k, destEmbeddingDelay, sourceHistoryEmbedLength,
				sourceEmbeddingDelay, delay);
	}

/**
	 * Initialise with (potentially) new parameters
	 * 
	 * @param base
	 * @param destHistoryEmbedLength
	 * @param destEmbeddingDelay
	 * @param sourceHistoryEmbeddingLength
	 * @param sourceEmbeddingDelay
	 * @param delay
	 */
	public void initialise(int base, int destHistoryEmbedLength, int destEmbeddingDelay,
			int sourceHistoryEmbeddingLength, int sourceEmbeddingDelay, int delay) {
		
		boolean paramsChanged = (this.base != base) || (k != destHistoryEmbedLength) ||
				(this.destEmbeddingDelay != destEmbeddingDelay) || (this.sourceHistoryEmbedLength != sourceHistoryEmbeddingLength) ||
				(this.sourceEmbeddingDelay != sourceEmbeddingDelay) || (this.delay != delay); 
		super.initialise(base, destHistoryEmbedLength);
		if (paramsChanged) {
			updateParameters(base, destHistoryEmbedLength, destEmbeddingDelay,
				sourceHistoryEmbeddingLength, sourceEmbeddingDelay, delay);
		}
		
		if (paramsChanged || (sourceNextPastCount == null)) {
			// Create new storage for extra counts of observations
			try {
				sourceNextPastCount = new int[base_power_l][base][base_power_k];
				sourcePastCount = new int[base_power_l][base_power_k];
			} catch (OutOfMemoryError e) {
				// Allow any Exceptions to be thrown, but catch and wrap
				//  Error as a RuntimeException
				throw new RuntimeException("Requested memory for the base " +
						base + ", k=" + k + ", l=" + sourceHistoryEmbedLength +
						" is too large for the JVM at this time", e);
			}
		} else {
			MatrixUtils.fill(sourceNextPastCount, 0);
			MatrixUtils.fill(sourcePastCount, 0);
		}
		estimateComputed = false;
	}
	

File: src/metrics/emergence/initialise.py
class MyClass:
    def __init__(self):
        self.base = None
        self.destHistoryEmbedLength = None
        self.destEmbeddingDelay = None
        self.sourceHistoryEmbedLength = None
        self.sourceEmbeddingDelay = None
        self.delay = None
        self.sourceNextPastCount = None
        self.sourcePastCount = None
        self.estimateComputed = False
        self.base_power_l = 0
        self.base_power_k = 0
        self.k = 0

    def initialise(self, base, destHistoryEmbedLength, destEmbeddingDelay,
                   sourceHistoryEmbeddingLength, sourceEmbeddingDelay, delay):
        paramsChanged = (self.base != base) or (self.k != destHistoryEmbedLength) or \
                        (self.destEmbeddingDelay != destEmbeddingDelay) or \
                        (self.sourceHistoryEmbedLength != sourceHistoryEmbeddingLength) or \
                        (self.sourceEmbeddingDelay != sourceEmbeddingDelay) or \
                        (self.delay != delay)
        
        # Assuming super.initialise(base, destHistoryEmbedLength) does some initialization
        # Here you should call the parent class initialise method if there is one
        
        if paramsChanged:
            self.updateParameters(base, destHistoryEmbedLength, destEmbeddingDelay,
                                  sourceHistoryEmbeddingLength, sourceEmbeddingDelay, delay)

        if paramsChanged or (self.sourceNextPastCount is None):
            try:
                self.sourceNextPastCount = [[[0 for _ in range(self.base_power_k)] for _ in range(base)] for _ in range(self.base_power_l)]
                self.sourcePastCount = [[0 for _ in range(self.base_power_k)] for _ in range(self.base_power_l)]
            except MemoryError as e:
                raise RuntimeError(f"Requested memory for the base {base}, k={self.k}, l={sourceHistoryEmbeddingLength} is too large for the system at this time") from e
        else:
            self.fill(self.sourceNextPastCount, 0)
            self.fill(self.sourcePastCount, 0)

        self.estimateComputed = False

    def updateParameters(self, base, destHistoryEmbedLength, destEmbeddingDelay,
                         sourceHistoryEmbeddingLength, sourceEmbeddingDelay, delay):
        # Implement the logic for updating the parameters here
        self.base = base
        self.k = destHistoryEmbedLength
        self.destEmbeddingDelay = destEmbeddingDelay
        self.sourceHistoryEmbedLength = sourceHistoryEmbeddingLength
        self.sourceEmbeddingDelay = sourceEmbeddingDelay
        self.delay = delay

    def fill(self, matrix, value):
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if isinstance(matrix[i][j], list):
                    self.fill(matrix[i][j], value)
                else:
                    matrix[i][j] = value

# Example usage
my_obj = MyClass()
my_obj.initialise(10, 5, 3, 7, 2, 4)
